File: forwardSubstitution.m
function z = forwardSubstitution(L, b)
    n = size(L, 1);
    z = zeros(n, 1);
    z(1) = b(1) / L(1, 1);
    for i = 2:n
        totalsum = 0;
        for j = 1 : i - 1
            totalsum = totalsum + L(i, j) * z(j);
        end
        z(i) = (b(i) - totalsum) / L(i, i);
    end
end

File: forwardSubstitution.py
# -*- coding: utf-8 -*-
"""
Created on Sat Feb  3 19:04:02 2024

@author: Kali Linux
"""

import numpy as np

def forwardSubstitution(L, b):
    n = L.shape[1]
    z = np.zeros((n, 1))
    
    z[0] = b[0] / L[0, 0]
    
    for i in range(1, n):
        totalsum = 0
        for j in range(0, i):
            totalsum += np.dot(L[i, j], z[j])
            
        z[i] = (b[i] - totalsum) / L[i, i]
        
    return z
